### ISSUE TITLE
[Rule change] `check-unnecessary-nullable` should ignore `Key? key`

### ISSUE THREAD

**Issue by fzyzcjy** [dart-code-checker/dart-code-metrics#974]
**What rule do you want to change?**

Sometimes, the widget's `key` argument is wrongly warned by check-unnecessary-nullable. Thus, we may need to check and ignore such cases, since by definition, `key` should be of type `Key?`

**How will the change be implemented? (New option, new default behavior, etc.)?**

**Please provide some example code that this change will affect:**

```dart
```

**What does the rule currently do for this code?**

**What will the rule do after it's changed?**

**Are you willing to submit a pull request to implement this change?**

**Comment by incendial** [comment]
@fzyzcjy could you provide a code example where is problem occurs to test it properly?

**Comment by fzyzcjy** [comment]
Sure, I will make a reproducible sample the next time I work on this linter

**Comment by incendial** [comment]
Thanks!

**Comment by fzyzcjy** [comment]
Seems that this happens everywhere.

```
class A extends StatelessWidget {
  const A({super.key}); // LINT
  ...
}

void main() {
  A(key: GlobalKey());
}
```

**Comment by incendial** [comment]
Thanks for the example, fixed. Waiting for release (as soon as we merge all the PR's).

**Comment by fzyzcjy** [comment]
Looking forward to the release!

**Comment by incendial** [comment]
Available in 4.18.0 🚀
